tools/elfdeps.cc

```diff
@@ -41,47 +44,38 @@ typedef struct elfInfo_s {
  * and common exception is the dynamic linker itself, which we allow
  * here, the rest can use --no-filter-soname.
  */
-static int skipSoname(const char *soname)
+static bool skipSoname(const std::string & soname)
 {
-    int sane = 0;
-
     /* Filter out empty and all-whitespace sonames */
-    for (const char *s = soname; *s != '\0'; s++) {
-	if (!risspace(*s)) {
-	    sane = 1;
-	    break;
-	}
-    }
+    if (soname.empty())
+	return true;
 
-    if (!sane)
-	return 1;
+    if (soname.find_first_not_of(" \t\n\r\f\v") == std::string::npos)
+	return true;
 
     if (filter_soname) {
-	if (!strstr(soname, ".so"))
-	    return 1;
-
-	if (rstreqn(soname, "ld.", 3) || rstreqn(soname, "ld-", 3) ||
-	    rstreqn(soname, "ld64.", 3) || rstreqn(soname, "ld64-", 3))
```

Comment:
Heh, there seems to have been a bug here (comparing at most `3` chars whereas the tested strings are longer). That's now fixed with the conversion :smile: Nice.

Reply:
Oh, look at that. I didn't notice... Funny timing as we just talked about humans and calculating string lengths yesterday :laughing: 

That bug exists in many 4.x releases then.

Reply:
Indeed, I've filed #3596 to cover the backports.